packages-exp/app-check-exp/src/types.ts

```diff
@@ -38,6 +38,12 @@ export interface AppCheckTokenObserver
   extends PartialObserver<AppCheckTokenResult> {
   // required
   next: AppCheckTokenListener;
+  type: ListenerType;
+}
+
+export enum ListenerType {
```

Comment:
Can be made const enum.

Reply:
Had to turn on `preserveConstEnums` in tsconfig to enable other files to import this as a const enum. Did this cause problems with `isolatedModules` in the past? It doesn't seem like it should, I checked the generated code and everything seems to be converted to literals, and we're not exporting this type publicly.

Reply:
I don't think it's needed for importing from other files in the same project. Auth doesn't turn on it. There is no point of changing to const enum if we have to turn on `preserveConstEnums`.

Reply:
Reverted tsconfig change. Rollup previously wouldn't build, but after moving the const enum to another file and then back, it built. I guess we can see in CI if it successfully builds from scratch.